java.lang.IllegalArgumentException: int is not compatible with java.lang.Integer in QBean

Using version 3.3.4 or later, I get an error when executing the following scala code:
```scala
object AddressBook extends SQL {
 ...
// setup SQL variables
...

def test = {
    QLists.all
}
}
```
This results in the exception:
Caused by: java.lang.IllegalArgumentException: int is not compatible with java.lang.Integer        
        at com.mysema.query.types.QBean.typeMismatch(QBean.java:248) ~[querydsl-core-3.3.4.jar:na] 
        at com.mysema.query.types.QBean.initFields(QBean.java:196) ~[querydsl-core-3.3.4.jar:na]   
        at com.mysema.query.types.QBean.<init>(QBean.java:176) ~[querydsl-core-3.3.4.jar:na]       
        at com.mysema.query.sql.RelationalPathUtils.createBeanProjection(RelationalPathUtils.java:61) ~[querydsl-sql-3.3.4.jar:na]
        at com.mysema.query.sql.RelationalPathUtils.createProjection(RelationalPathUtils.java:41) ~[querydsl-sql-3.3.4.jar:na]
        at com.mysema.query.scala.sql.RelationalPathImpl.projection$lzycompute(Paths.scala:50) ~[querydsl-scala-3.3.4.jar:na]
        at com.mysema.query.scala.sql.RelationalPathImpl.projection(Paths.scala:50) ~[querydsl-scala-3.3.4.jar:na]
        at com.mysema.query.scala.sql.RelationalPathImpl.getProjection(Paths.scala:50) ~[querydsl-scala-3.3.4.jar:na]
        at com.mysema.query.scala.sql.RelationalPathImpl.getProjection(Paths.scala:33) ~[querydsl-scala-3.3.4.jar:na]
        at com.mysema.query.support.QueryMixin.convert(QueryMixin.java:139) ~[querydsl-core-3.3.4.jar:na]
        at com.mysema.query.support.QueryMixin.addProjection(QueryMixin.java:105) ~[querydsl-core-3.3.4.jar:na]
        at com.mysema.query.sql.AbstractSQLQuery.list(AbstractSQLQuery.java:231) ~[querydsl-sql-3.3.4.jar:na]
        at com.mysema.query.scala.RichProjectable.select(Helpers.scala:64) ~[querydsl-scala-3.3.4.jar:na]
        at com.mysema.query.scala.sql.RichSimpleQuery.select(Queries.scala:72) ~[querydsl-scala-3.3.4.jar:na]
        at com.mysema.query.scala.sql.RichSimpleQuery.all(Queries.scala:68) ~[querydsl-scala-3.3.4.jar:na]
        at net.globalrecordings.address_book.AddressBook$.testQueryDSLbug(AddressBook.scala:54) ~[classes/:na]
        at net.globalrecordings.address_book.AddressBook$$anonfun$2$$anonfun$apply$2.apply(AddressBook.scala:37) ~[classes/:na]
        at net.globalrecordings.address_book.AddressBook$$anonfun$2$$anonfun$apply$2.apply(AddressBook.scala:28) ~[classes/:na]
        at net.liftweb.common.Full.foreach(Box.scala:551) ~[lift-common_2.10-2.5.3.jar:2.5.3]
        at net.globalrecordings.address_book.AddressBook$$anonfun$2.apply(AddressBook.scala:28) ~[classes/:na]
        at net.globalrecordings.address_book.AddressBook$$anonfun$2.apply(AddressBook.scala:27) ~[classes/:na]
        at net.liftweb.common.Full.foreach(Box.scala:551) ~[lift-common_2.10-2.5.3.jar:2.5.3]
        at net.globalrecordings.address_book.AddressBook$.<init>(AddressBook.scala:27) ~[classes/:na]
        at net.globalrecordings.address_book.AddressBook$.<clinit>(AddressBook.scala) ~[classes/:na]
        ... 312 common frames omitted


This error does not occur in version 3.3.3, everything runs fine. So I know the surrounding code is working fine. After doing some searching, it looks like the change in commit 
79c9d4335e35b6769a84df67bdc7810f448a4885 in issue #710  (merge #761) is what has broken this. It some how is no longer matching "int" to "java.lang.Integer". The generated  scala code for QLists and Lists is below. It does not change between 3.3.3 and 3.3.4 though. The error occurs in version 3.7.0 as well.

Please let me know if you need anything else to help fix this issue. Thanks.

Kevin Horan



```scala
package net.globalrecordings.gr_database.address_book                                                                   
                                                                         
object Lists extends QLists("lists") {                                                                                  
  override def as(variable: String) = new QLists(variable)                                                              
                                                                                                                        
}                                                                                                                       
                                                                                                                        
/**                                                                                                                     
 * Lists is a Querydsl bean type                                                                                        
 */                                                                                                                     
class Lists {                                                                                                           
                                                                                                                        
  var listKey: Integer = _                                                                                              
                                                                                                                        
  var listName: String = _                                                                                              
                                                                                                                        
  var listOwner: Integer = _                                                                                            
                                                                                                                        
  var notes: String = _                                                                                                 
                                                                                                                        
}               
```

```scala
package net.globalrecordings.gr_database.address_book                                                                      
                                                                                                                           
import com.mysema.query.types._                                                                                            
import com.mysema.query.scala._                                                                                            
                                                                                                                           
import com.mysema.query.types.PathMetadataFactory._;                                                                       
                                                                                                                           
import com.mysema.query.scala.sql.RelationalPathImpl                                                                       
                                                                                                                           
import com.mysema.query.sql._                                                                                              
                                                                                                                           
object QLists extends QLists("lists") {                                                                                    
  override def as(variable: String) = new QLists(variable)                                                                 
                                                                                                                           
}                                                                                                                          
                                                                                                                           
class QLists(md: PathMetadata[_]) extends RelationalPathImpl[Lists](md, "address_book", "lists") {                         
  def this(variable: String) = this(forVariable(variable))                                                                 
                                                                                                                           
  def this(parent: Path[_], property: String) = this(forProperty(parent, property))                                        
                                                                                                                           
  val listKey = createNumber[Int]("listKey")                                                                               
                                                                                                                           
  val listName = createString("listName")                                                                               
                                                                                                                        
  val listOwner = createNumber[Int]("listOwner")                                                                        
                                                                                                                        
  val notes = createString("notes")                                                                                     
                                                                                                                        
  val listsPkey: PrimaryKey[Lists] = createPrimaryKey(listKey)                                                          
                                                                                                                        
  val namesFkey: ForeignKey[Names] = createForeignKey(listOwner, "name_key")                                            
                                                                                                                        
  val _listsFkey: ForeignKey[ListMembers] = createInvForeignKey(listKey, "list_key")                                    
                                                                                                                        
  val _groupsOfListsListKeyFkey: ForeignKey[GroupsOfLists] = createInvForeignKey(listKey, "list_key")                   
                                                                                                                        
  addMetadata(listKey, ColumnMetadata.named("list_key").ofType(4).withSize(10).notNull())                               
  addMetadata(listName, ColumnMetadata.named("list_name").ofType(12).withSize(127).notNull())                           
  addMetadata(listOwner, ColumnMetadata.named("list_owner").ofType(4).withSize(10).notNull())                           
  addMetadata(notes, ColumnMetadata.named("notes").ofType(12).withSize(2147483647).notNull())                           
}                                                                                                                       
     
```

--- a/querydsl-core/src/main/java/com/querydsl/core/types/QBean.java
+++ b/querydsl-core/src/main/java/com/querydsl/core/types/QBean.java
@@ -86,6 +86,10 @@ public class QBean<T> extends FactoryExpressionBase<T> {
         return cl.isPrimitive() ? Primitives.wrap(cl) : cl;
     }
 
+    private static boolean isAssignableFrom(Class<?> cl1, Class<?> cl2) {
+        return normalize(cl1).isAssignableFrom(normalize(cl2));
+    }
+
     private final ImmutableMap<String, Expression<?>> bindings;
 
     private final List<Field> fields;
@@ -156,7 +160,7 @@ public class QBean<T> extends FactoryExpressionBase<T> {
                 try {
                     field = beanType.getDeclaredField(property);
                     field.setAccessible(true);
-                    if (!normalize(field.getType()).isAssignableFrom(expr.getType())) {
+                    if (!isAssignableFrom(field.getType(), expr.getType())) {
                         typeMismatch(field.getType(), expr);
                     }
                     beanType = Object.class;
@@ -186,7 +190,7 @@ public class QBean<T> extends FactoryExpressionBase<T> {
                 for (PropertyDescriptor prop : propertyDescriptors) {
                     if (prop.getName().equals(property)) {
                         setter = prop.getWriteMethod();
-                        if (!normalize(prop.getPropertyType()).isAssignableFrom(expr.getType())) {
+                        if (!isAssignableFrom(prop.getPropertyType(), expr.getType())) {
                             typeMismatch(prop.getPropertyType(), expr);
                         }
                         break;

